server/routes/charts.js

```diff
@@ -39,9 +40,45 @@ router.route('/charts/new')
 router.route('/charts/:chart_id')
   .get(ensureAuthenticated, async(req,res) => {
     try {
+      const data = []
+      const { dataDb, appDb } = req.app.locals
       const { chart_id } = req.params
+      const { access } =  await appDb
+        .collection(collections.users)
+        .findOne(
+          { uid: req.user },
+          { _id: 0, access: 1 }
+        )
+      const { assessment, variable, fieldLabelValueMap, title } = await dataDb
+        .collection(collections.charts)
+        .findOne({ _id: new ObjectID(chart_id) })
+
+      for await (const field of fieldLabelValueMap) {
+        const valueCountData = []
+
+        for await (const study of access ){
+          let count = 0;
+          const { label, value } = field;
```

Comment:
can we move this up above this nested `for` so we're not re-creating these constants for each study?

Reply:
[3d5d8c](https://github.com/AMP-SCZ/dpdash/pull/91/commits/3d5d8c2367dffb4b84223b17754ce6f1a2ecb5c6)